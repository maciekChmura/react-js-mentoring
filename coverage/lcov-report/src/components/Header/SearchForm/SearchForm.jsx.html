<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Header/SearchForm/SearchForm.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/Header/SearchForm</a> SearchForm.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import {
  Wrapper,
  Display,
  InputWrapper,
  ButtonsWrapper,
  SearchButton
} from './SearchForm.Styles';
import OptionButton from '../../Helper/OptionButton/OptionButton';
&nbsp;
class SearchForm extends Component {
  state = {
    searchValue: ''
  };
&nbsp;
  handleInputChange = <span class="fstat-no" title="function not covered" >ev</span>ent =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ searchValue: event.target.value });</span>
  };
&nbsp;
  handleKeyPress = <span class="fstat-no" title="function not covered" >ev</span>ent =&gt; {
<span class="cstat-no" title="statement not covered" >    if (event.key === 'Enter') {</span>
<span class="cstat-no" title="statement not covered" >      this.handleFormSubmit();</span>
    }
  };
&nbsp;
  handleFormSubmit = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.props.handleFormSubmit(this.state.searchValue);</span>
  };
&nbsp;
  render() {
    const { searchOption, changeSearch } = this.props;
    const { searchValue } = this.state;
    return (
      &lt;Wrapper&gt;
        &lt;form onSubmit={this.handleFormSubmit}&gt;
          &lt;InputWrapper
            id="location"
            placeholder="type something"
            value={searchValue}
            onChange={this.handleInputChange}
            onKeyPress={this.handleKeyPress}
          /&gt;
&nbsp;
          &lt;ButtonsWrapper&gt;
            &lt;Display&gt;Search by: &lt;/Display&gt;
            {['title', 'genres'].map(title =&gt; (
              &lt;OptionButton
                text={title}
                changeOption={changeSearch}
                option={searchOption}
                key={title}
              /&gt;
            ))}
            &lt;SearchButton onClick={this.handleFormSubmit} type="button"&gt;
              Search
            &lt;/SearchButton&gt;
          &lt;/ButtonsWrapper&gt;
        &lt;/form&gt;
      &lt;/Wrapper&gt;
    );
  }
}
&nbsp;
export default SearchForm;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 21 2018 15:25:26 GMT+0100 (STD)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
