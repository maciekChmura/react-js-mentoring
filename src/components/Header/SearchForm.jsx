import React, { Component } from 'react';
import styled from 'styled-components';
import OptionButton from '../Helper/OptionButton';

const Wrapper = styled.div`
  grid-area: 3 / 2 / 3 / 4;
  display: flex;
  flex-direction: column;
`;

const InputWrapper = styled.input`
  font-size: 18px;
  border: 0;
  width: 780px;
  height: 24px;
  background: #fff;
  padding: 6px;
  border: 1px solid grey;
  border-radius: 4px;
`;

const ButtonsWrapper = styled.div`
  width: 800px;
  display: flex;
  margin-top: 10px;
`;

const Display = styled.p`
  font-size: 18px;
  margin: 5px;
  padding: 5px;
  color: white;
`;

const SearchButton = styled.button`
  background-color: #dd1133;
  height: 32px;
  border: 0;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  margin: 4px 10px 4px 10px;
  padding: 2px 10px 2px 10px;
  margin-left: auto;
`;

class SearchForm extends Component {
  state = {
    searchValue: ''
  };

  handleInputChange = event => {
    this.setState({ searchValue: event.target.value });
  };

  handleKeyPress = event => {
    if (event.key === 'Enter') {
      this.handleFormSubmit();
    }
  };

  handleFormSubmit = () => {
    event.preventDefault();
    this.props.handleFormSubmit(this.state.searchValue);
  };

  render() {
    const { searchOption, changeSearch } = this.props;
    const { searchValue } = this.state;
    return (
      <Wrapper>
        <form onSubmit={this.handleFormSubmit}>
          <InputWrapper
            id="location"
            placeholder="type something"
            value={searchValue}
            onChange={this.handleInputChange}
            onKeyPress={this.handleKeyPress}
          />

          <ButtonsWrapper>
            <Display>Search by: </Display>
            <OptionButton
              text="title"
              changeOption={changeSearch}
              option={searchOption}
            />
            <OptionButton
              text="genres"
              changeOption={changeSearch}
              option={searchOption}
            />
            <SearchButton onClick={this.handleFormSubmit} type="button">
              Search
            </SearchButton>
          </ButtonsWrapper>
        </form>
      </Wrapper>
    );
  }
}

export default SearchForm;
