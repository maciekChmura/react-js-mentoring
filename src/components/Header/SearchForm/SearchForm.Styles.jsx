import styled from 'styled-components';

export const Wrapper = styled.div`
  grid-area: 3 / 2 / 3 / 4;
  display: flex;
  flex-direction: column;
`;

export const InputWrapper = styled.input`
  font-size: 18px;
  border: 0;
  width: 780px;
  height: 24px;
  background: #fff;
  padding: 6px;
  border: 1px solid grey;
  border-radius: 4px;
`;

export const ButtonsWrapper = styled.div`
  width: 800px;
  display: flex;
  margin-top: 10px;
`;

export const Display = styled.p`
  font-size: 18px;
  margin: 5px;
  padding: 5px;
  color: white;
`;

export const SearchButton = styled.button`
  background-color: #dd1133;
  height: 32px;
  border: 0;
  border-radius: 4px;
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  margin: 4px 10px 4px 10px;
  padding: 2px 10px 2px 10px;
  margin-left: auto;
`;
