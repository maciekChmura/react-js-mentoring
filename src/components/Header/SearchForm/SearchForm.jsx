import React, { Component } from 'react';
import { connect } from 'react-redux';
import { changeSearch, updateSearchValue } from '../../../redux/actions';
import {
  Wrapper,
  Display,
  InputWrapper,
  ButtonsWrapper,
  SearchButton
} from './SearchForm.Styles';
import OptionButton from '../../Helper/OptionButton/OptionButton';

const mapStateToProps = state => ({
  searchOption: state.searchOption,
  searchValue: state.searchValue
});

export class SearchForm extends Component {
  state = {
    stateSearchValue: ''
  };

  handleInputChange = event => {
    this.setState({ stateSearchValue: event.target.value });
  };

  handleKeyPress = event => {
    if (event.key === 'Enter') {
      this.props.handleFormSubmit(this.state.stateSearchValue);
    }
  };

  render() {
    const { searchOption, handleFormSubmit, changeSearch } = this.props;
    const { stateSearchValue } = this.state;
    return (
      <Wrapper>
        <form onSubmit={() => handleFormSubmit(stateSearchValue)}>
          <InputWrapper
            id="location"
            placeholder="type something"
            value={stateSearchValue}
            onChange={this.handleInputChange}
            onKeyPress={this.handleKeyPress}
          />
          <ButtonsWrapper>
            <Display>Search by: </Display>
            {['title', 'genres'].map(title => (
              <OptionButton
                text={title}
                changeOption={() => changeSearch(title)}
                option={searchOption}
                key={title}
              />
            ))}
            <SearchButton
              onClick={() => handleFormSubmit(stateSearchValue)}
              type="button"
            >
              Search
            </SearchButton>
          </ButtonsWrapper>
        </form>
      </Wrapper>
    );
  }
}

export default connect(
  mapStateToProps,
  { changeSearch, updateSearchValue }
)(SearchForm);
