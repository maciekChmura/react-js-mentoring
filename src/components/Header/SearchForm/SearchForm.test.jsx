import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchForm />);

    expect(component).toMatchSnapshot();
  });

  it('InputWrapper onChange updates state', () => {
    const component = shallow(<SearchForm />);
    const eventData = {
      target: {
        value: 'test'
      }
    };
    component.find('InputWrapper').prop('onChange')(eventData);
    expect(component.state('searchValue')).toEqual('test');
  });

  it('InputWrapper onKeyPress calls handleFormSubmit', () => {
    const component = shallow(<SearchForm />);
    const eventData = {
      key: 'Enter'
    };

    const handleFormSubmitMock = jest.fn();
    component.instance().handleFormSubmit = handleFormSubmitMock;
    component.update();
    component.instance().handleKeyPress(eventData);
    expect(handleFormSubmitMock).toBeCalled();
  });
});
