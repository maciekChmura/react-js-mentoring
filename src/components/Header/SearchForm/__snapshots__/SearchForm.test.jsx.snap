// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <ForwardRef
          id="location"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="type something"
          value=""
        />
        <ForwardRef>
          <ForwardRef>
            Search by: 
          </ForwardRef>
          <OptionButton
            text="title"
          />
          <OptionButton
            text="genres"
          />
          <ForwardRef
            onClick={[Function]}
            type="button"
          >
            Search
          </ForwardRef>
        </ForwardRef>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef
            id="location"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="type something"
            value=""
          />,
          <ForwardRef>
            <ForwardRef>
              Search by: 
            </ForwardRef>
            <OptionButton
              text="title"
            />
            <OptionButton
              text="genres"
            />
            <ForwardRef
              onClick={[Function]}
              type="button"
            >
              Search
            </ForwardRef>
          </ForwardRef>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "location",
            "onChange": [Function],
            "onKeyPress": [Function],
            "placeholder": "type something",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "SearchFormStyles__InputWrapper-emg76u-1",
              "isStatic": true,
              "rules": Array [
                "font-size:18px;border:0;width:780px;height:24px;background:#fff;padding:6px;border:1px solid grey;border-radius:4px;",
              ],
            },
            "displayName": "InputWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "SearchFormStyles__InputWrapper-emg76u-1",
            "target": "input",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                Search by: 
              </ForwardRef>,
              Array [
                <OptionButton
                  text="title"
                />,
                <OptionButton
                  text="genres"
                />,
              ],
              <ForwardRef
                onClick={[Function]}
                type="button"
              >
                Search
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Search by: ",
              },
              "ref": null,
              "rendered": "Search by: ",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchFormStyles__Display-emg76u-3",
                  "isStatic": true,
                  "rules": Array [
                    "font-size:18px;margin:5px;padding:5px;color:white;",
                  ],
                },
                "displayName": "SearchFormStyles__Display",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchFormStyles__Display-emg76u-3",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": "title",
              "nodeType": "function",
              "props": Object {
                "changeOption": undefined,
                "option": undefined,
                "text": "title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "genres",
              "nodeType": "function",
              "props": Object {
                "changeOption": undefined,
                "option": undefined,
                "text": "genres",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Search",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Search",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "SearchFormStyles__SearchButton-emg76u-4",
                  "isStatic": true,
                  "rules": Array [
                    "background-color:#dd1133;height:32px;border:0;border-radius:4px;font-size:14px;font-weight:600;text-transform:uppercase;margin:4px 10px 4px 10px;padding:2px 10px 2px 10px;margin-left:auto;",
                  ],
                },
                "displayName": "SearchFormStyles__SearchButton",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "SearchFormStyles__SearchButton-emg76u-4",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "SearchFormStyles__ButtonsWrapper-emg76u-2",
              "isStatic": true,
              "rules": Array [
                "width:800px;display:flex;margin-top:10px;",
              ],
            },
            "displayName": "SearchFormStyles__ButtonsWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "SearchFormStyles__ButtonsWrapper-emg76u-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "form",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "SearchFormStyles__Wrapper-emg76u-0",
        "isStatic": true,
        "rules": Array [
          "grid-area:3 / 2 / 3 / 4;display:flex;flex-direction:column;",
        ],
      },
      "displayName": "SearchFormStyles__Wrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "SearchFormStyles__Wrapper-emg76u-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <ForwardRef
            id="location"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="type something"
            value=""
          />
          <ForwardRef>
            <ForwardRef>
              Search by: 
            </ForwardRef>
            <OptionButton
              text="title"
            />
            <OptionButton
              text="genres"
            />
            <ForwardRef
              onClick={[Function]}
              type="button"
            >
              Search
            </ForwardRef>
          </ForwardRef>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef
              id="location"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="type something"
              value=""
            />,
            <ForwardRef>
              <ForwardRef>
                Search by: 
              </ForwardRef>
              <OptionButton
                text="title"
              />
              <OptionButton
                text="genres"
              />
              <ForwardRef
                onClick={[Function]}
                type="button"
              >
                Search
              </ForwardRef>
            </ForwardRef>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "location",
              "onChange": [Function],
              "onKeyPress": [Function],
              "placeholder": "type something",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "SearchFormStyles__InputWrapper-emg76u-1",
                "isStatic": true,
                "rules": Array [
                  "font-size:18px;border:0;width:780px;height:24px;background:#fff;padding:6px;border:1px solid grey;border-radius:4px;",
                ],
              },
              "displayName": "InputWrapper",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "SearchFormStyles__InputWrapper-emg76u-1",
              "target": "input",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef>
                  Search by: 
                </ForwardRef>,
                Array [
                  <OptionButton
                    text="title"
                  />,
                  <OptionButton
                    text="genres"
                  />,
                ],
                <ForwardRef
                  onClick={[Function]}
                  type="button"
                >
                  Search
                </ForwardRef>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Search by: ",
                },
                "ref": null,
                "rendered": "Search by: ",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "SearchFormStyles__Display-emg76u-3",
                    "isStatic": true,
                    "rules": Array [
                      "font-size:18px;margin:5px;padding:5px;color:white;",
                    ],
                  },
                  "displayName": "SearchFormStyles__Display",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "SearchFormStyles__Display-emg76u-3",
                  "target": "p",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": "title",
                "nodeType": "function",
                "props": Object {
                  "changeOption": undefined,
                  "option": undefined,
                  "text": "title",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "genres",
                "nodeType": "function",
                "props": Object {
                  "changeOption": undefined,
                  "option": undefined,
                  "text": "genres",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Search",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Search",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "SearchFormStyles__SearchButton-emg76u-4",
                    "isStatic": true,
                    "rules": Array [
                      "background-color:#dd1133;height:32px;border:0;border-radius:4px;font-size:14px;font-weight:600;text-transform:uppercase;margin:4px 10px 4px 10px;padding:2px 10px 2px 10px;margin-left:auto;",
                    ],
                  },
                  "displayName": "SearchFormStyles__SearchButton",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "SearchFormStyles__SearchButton-emg76u-4",
                  "target": "button",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "SearchFormStyles__ButtonsWrapper-emg76u-2",
                "isStatic": true,
                "rules": Array [
                  "width:800px;display:flex;margin-top:10px;",
                ],
              },
              "displayName": "SearchFormStyles__ButtonsWrapper",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "SearchFormStyles__ButtonsWrapper-emg76u-2",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": "form",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "SearchFormStyles__Wrapper-emg76u-0",
          "isStatic": true,
          "rules": Array [
            "grid-area:3 / 2 / 3 / 4;display:flex;flex-direction:column;",
          ],
        },
        "displayName": "SearchFormStyles__Wrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "SearchFormStyles__Wrapper-emg76u-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
